src/AzureClient/Magic/AzureClientMagicBase.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Quantum.IQSharp.Common;
+using Microsoft.Quantum.IQSharp.Jupyter;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    /// <summary>
+    ///     Base class used for Azure Client magic commands.
+    /// </summary>
+    public abstract class AzureClientMagicBase : AbstractMagic
+    {
+        /// <summary>
+        ///     The object used by this magic command to interact with Azure.
+        /// </summary>
+        public IAzureClient AzureClient { get; }
+
+        /// <summary>
+        ///     Constructs the Azure Client magic command with the specified keyword
+        ///     and documentation.
+        /// </summary>
+        public AzureClientMagicBase(IAzureClient azureClient, string keyword, Documentation docs):
+            base(keyword, docs)
+        {
+            this.AzureClient = azureClient;
```

Comment:
could be null

Reply:
In general, the magic command classes receive their arguments from the `DependencyInjection` infrastructure (see `Startup.ConfigureServices()` in the `Tool` project), which keeps singletons of many objects, including `AzureClient`, and automatically provides references to those singletons when constructing the magic commands.

All that said, in general if this is null then something very unexpected went wrong at startup, so throwing an exception in that case would be fine. In general, IQ# code doesn't null-check the parameters coming from the `DependencyInjection` framework, because they should never be null.

---
In reply to: [426994603](https://github.com/microsoft/iqsharp/pull/144#discussion_r426994603) [](ancestors = 426994603)